src/pilot/assign_role.py

```diff
@@ -888,17 +921,6 @@ def assign_role(ip_mac_service_tag, node_uuid, role_index, os_volume_size_gb,
 def generate_osd_config(ip_mac_service_tag, drac_client):
     controllers = drac_client.list_raid_controllers()
 
-    found_hba = False
-    for controller in controllers:
-        if "hba330" in controller.model.lower():
-            found_hba = True
-            break
-
-    if not found_hba:
-        LOG.info("Not generating OSD config for {ip} because no HBA330 is "
-                 "present.".format(ip=ip_mac_service_tag))
-        return
-
     LOG.info("Generating OSD config for {ip}".format(ip=ip_mac_service_tag))
     system_id = drac_client.get_system().uuid
 
```

Comment:
Does the below work with a PERC H740P RAID controller?  The reason why I ask is that we will create single disk RAID0s in the RAID configuration section, but the get_drives() method below only deals with physical disks and not single disk RAID0s.  I would think get_drives would need to return the single disk RAID0s in this case and not the physical disks that back them.

Reply:
As discussed in technical call, we have tested the below code with  PERC H740P RAID controller and code is working fine.
We are able to get osd's for  PERC H740P RAID controller with the below code.